### PR TITLE
Feature/include api rating

### LINKED ISSUE #1618: Include star rating widget in API profile and Catalog
Reintroduce star rating into platform. Needs to be visible and editable in API catalog and API profile.

# Definition of done
* [x] Bookmarking icon and terminology needs to be distinct from star rating
  * [x] Bookmarking is called bookmark, icon can be freely chosen
  * [x] Rating uses star metaphor in icons, terminology refers to rating
* [x] Star rating is visible in Catalog (both card and table view)
* [x] Star rating is visible in API profile
* [ ] User can add and edit their rating 
  * [ ] in Catalog  (both card and table view)
  * [x] in API profile

# Wireframes

1. API Card View

apicardnewbookmark-1

2. API Table View

apitablenew

3. API Profile

apiprofilestar bookmark

### PR DESCRIPTION
**Pull request by nnn**
Closes #1618 

## Proposed changes
- rating at api profile page
- rating at table view of catalog
- average rating at grid view of catalog
- restructure bookmark collection to match other collections

### CODE REVIEW

**Comment by nnn** [comment]
@apinf/developers, please review.

**Comment by bajiat** [comment]
@marla-singer Would you be available to review this PR?

**Comment by marla-singer** [comment]
@bajiat Yes

**Comment by bajiat** [comment]
@marla-singer Thanks for taking the review task!

**Comment by marla-singer** [comment]
@nnn "My bookmarks" filter doesn't work in catalog.
Screenshots:

1. Add api to bookmarks
joxi_screenshot_1475673356699

2. Filtering
joxi_screenshot_1475673386737

Maybe it happens because it isn't subscribed to Bookmarks collection when user clicks on button "My bookmarks". See screenshot below

**Comment by nnn** [comment]
@marla-singer fixed.

File: bookmarks/server/methods.js [hunk 1]
```diff
@@ -1,27 +1,28 @@
 import { Apis } from '/apis/collection';
+import { ApiBookmarks } from '/bookmarks/collection';
 
 Meteor.methods({
-  "toggleBookmarkApi": function (backendId, currentUserId) {
+  'toggleBookmarkApi': function (backendId, currentUserId) {
```
Comment (marla-singer) [inline bookmarks/server/methods.js]: Rewrite this line using ES2015 standard too

File: bookmarks/client/bookmark.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
   // Get reference to template instance
   const instance = this;
 
   // subscribe to user bookmarks, creating reference to subscription
   instance.bookmarksSubscription = instance.subscribe('myApiBookmarks');
 };
 
-Template.favourite.events({
+Template.bookmark.events({
   'click .bookmark': function () {
```
Comment (marla-singer) [inline bookmarks/client/bookmark.js]: Rewrite this line using ES2015 standard

File: bookmarks/collection/server/publications.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { ApiBookmarks } from '/bookmarks/collection';
+import { Meteor } from 'meteor/meteor';
+
```
Comment (marla-singer) [inline bookmarks/collection/server/publications.js]: Change order: at first is meteor packages import after collections import

File: bookmarks/client/bookmark.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<template name="bookmark">
+  {{ # if isBookmarked }}
```
Comment (marla-singer) [inline bookmarks/client/bookmark.html]: Now Brylie has done a global code style refactoring. Please change the code style in this file. Look the standards on this page [https://github.com/apinf/platform/issues/1690](url)

**Comment by marla-singer** [comment]
@nnn Please fix conflicts

**Comment by nnn** [comment]
@marla-singer please review. Conflicts resolved.

File: bookmarks/client/bookmark.js (hunk 2 above)
Comment (brylie) [inline bookmarks/client/bookmark.js]: The ES2015 function would look like this:

```js
Template.bookmark.events({
  'click .bookmark' () {
    // Do something
  }
});
```

**Comment by marla-singer** [comment]
@nnn The problem `Exception while invoking method 'toggleBookmarkApi' ReferenceError: apiIds is not defined` exists yet.

**Comment by nnn** [comment]
@marla-singer, that's weird. Didn't get this error before. Checking.

**Comment by nnn** [comment]
@marla-singer please review.

**Comment by marla-singer** [comment]
@nnn Sorry for delay. Works correctly. Going to merge it

**Pull request merged**